Lib/test/test_typing.py

```diff
@@ -185,6 +185,8 @@ def test_union_unique(self):
         self.assertEqual(Union[X, int].__args__, (X, int))
         self.assertEqual(Union[X, int].__parameters__, (X,))
         self.assertIs(Union[X, int].__origin__, Union)
+        self.assertEqual(X | "x", Union[X, "x"])
+        self.assertEqual("x" | X, Union["x", X])
```

Comment:
I don't think these are testing for what Larry meant, Union comparisons are order independent, so
```python
>>> T | "x" == "x" | T
True
>>> T | "x" == Union["x" | T]
True
```
What he meant is that the ``repr`` _is_ order dependent, so the test should be like:
```suggestion
        self.assertEqual(X | "x", repr(X | "x")
        self.assertEqual("x" | X, repr("x" | X)
```
Which currently fails without the patch.

Reply:
Good point, I should have written the test first to make sure I was actually fixing anything.

Reply:
Made it use `get_args()` to test though, which feels cleaner than testing the `repr()` directly.

Reply:
Yeah I think that works :). The ``__repr__`` is dependent on ``__args__`` after all.